src/test/java/io/neonbee/test/endpoint/odata/ODataNavigationPropertiesTest.java

```diff
@@ -0,0 +1,104 @@
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.PRODUCTS_ENTITY_SET_FQN;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.PROPERTY_NAME_CATEGORY;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.PROPERTY_NAME_CATEGORY_ID;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.PROPERTY_NAME_ID;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.STEAK_PRODUCT;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.STREET_GLIDE_SPECIAL_PRODUCT;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.S_1000_RR_PRODUCT;
+import static io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle.getDeclaredEntityModel;
+import static java.util.stream.Collectors.toList;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.neonbee.test.base.ODataEndpointTestBase;
+import io.neonbee.test.base.ODataRequest;
+import io.neonbee.test.endpoint.odata.verticle.NavPropsCategoriesEntityVerticle;
+import io.neonbee.test.endpoint.odata.verticle.NavPropsProductsEntityVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * These tests cover the expand option which can be declared with the following kind of request:<br>
+ * <br>
+ *
+ * <pre>
+ * http://baseUrl/odata/io.neonbee.test.NavProbs/Categories(2)/products
+ * </pre>
+ *
+ */
+public class ODataNavigationPropertiesTest extends ODataEndpointTestBase {
+    @Override
+    protected List<Path> provideEntityModels() {
+        return List.of(getDeclaredEntityModel());
+    }
+
+    @BeforeEach
+    void setUp(VertxTestContext testContext) {
+        CompositeFuture
+                .all(deployVerticle(new NavPropsProductsEntityVerticle()),
+                        deployVerticle(new NavPropsCategoriesEntityVerticle()))
+                .onComplete(testContext.succeedingThenComplete());
+    }
+
+    static Stream<Arguments> withProducts() {
+        return Stream.of(Arguments.of(STEAK_PRODUCT, FOOD_CATEGORY), Arguments.of(CHEESE_PRODUCT, FOOD_CATEGORY),
+                Arguments.of(S_1000_RR_PRODUCT, MOTORCYCLE_CATEGORY),
+                Arguments.of(STREET_GLIDE_SPECIAL_PRODUCT, MOTORCYCLE_CATEGORY));
+    }
+
+    @ParameterizedTest(name = "{index}: For Product: {0}")
+    @MethodSource("withProducts")
+    @Timeout(value = 2, timeUnit = TimeUnit.HOURS)
```

Comment:
I would take `TimeUnit.SECONDS` instead of `TimeUnit.HOURS` :)

Reply:
Mh could you explain why? :)